frontend/www/js/omegaup/components/qualitynomination/Popup.vue

```diff
@@ -272,3 +171,138 @@ label.tag-select:hover {
+import UI from '../../ui.js';
+
+interface ProblemTopic {
+  text: string;
+  value: string;
+}
+
+@Component
+export default class QualityNominationPopup extends Vue {
+  @Prop() solved!: boolean;
+  @Prop() nominated!: boolean;
+  @Prop() dismissed!: boolean;
+  @Prop() canNominateProblem!: boolean;
+
+  T = T;
+  UI = UI;
+  currentView = 'suggestion';
+  difficulty = '';
+  quality = '';
+  showFormOverride = true;
+  localDismissed = this.dismissed;
+  localNominated = this.nominated;
+  tags: string[] = [];
+  static readonly PROBLEM_TOPICS = [
+    'problemTopic2Sat',
+    'problemTopicArrays',
+    'problemTopicBacktracking',
+    'problemTopicBigNumbers',
+    'problemTopicBinarySearch',
+    'problemTopicBitmasks',
+    'problemTopicBreadthDepthFirstSearch',
+    'problemTopicBruteForce',
+    'problemTopicBuckets',
+    'problemTopicCombinatorics',
+    'problemTopicDataStructures',
+    'problemTopicDisjointSets',
+    'problemTopicDivideAndConquer',
+    'problemTopicDynamicProgramming',
+    'problemTopicFastFourierTransform',
+    'problemTopicGameTheory',
+    'problemTopicGeometry',
+    'problemTopicGraphTheory',
+    'problemTopicGreedy',
+    'problemTopicHashing',
+    'problemTopicIfElseSwitch',
+    'problemTopicImplementation',
+    'problemTopicInputOutput',
+    'problemTopicLoops',
+    'problemTopicMath',
+    'problemTopicMatrices',
+    'problemTopicMaxFlow',
+    'problemTopicMeetInTheMiddle',
+    'problemTopicNumberTheory',
+    'problemTopicParsing',
+    'problemTopicProbability',
+    'problemTopicShortestPath',
+    'problemTopicSimulation',
+    'problemTopicSorting',
+    'problemTopicStackQueue',
+    'problemTopicStrings',
+    'problemTopicSuffixArray',
+    'problemTopicSuffixTree',
+    'problemTopicTernarySearch',
+    'problemTopicTrees',
+    'problemTopicTwoPointers',
+  ];
+
+  get showForm(): boolean {
+    return (
+      this.showFormOverride &&
+      this.solved &&
+      !this.localNominated &&
+      !this.localDismissed &&
+      this.canNominateProblem
+    );
+  }
+
+  get showSuggestLink(): boolean {
+    return this.solved && !this.localNominated;
+  }
+
+  get sortedProblemTopics(): ProblemTopic[] {
+    let self = this;
+    let topics: ProblemTopic[] = QualityNominationPopup.PROBLEM_TOPICS.map(
+      (x: string) => {
+        return {
+          value: x,
+          text: self.T[x],
+        };
+      },
+    );
+    return topics.sort((a: ProblemTopic, b: ProblemTopic): number => {
+      return a.text.localeCompare(b.text, self.T.lang);
+    });
+  }
+
+  onHide(isDismissed: boolean): void {
+    this.showFormOverride = false;
+    if (isDismissed) {
+      this.$emit('dismiss', this);
+    }
+  }
+
+  onShowSuggestion(): void {
+    this.showFormOverride = true;
+    this.localDismissed = false;
+  }
+
+  onSubmit(): void {
+    this.$emit('submit', this);
+    this.currentView = 'thanks';
+    this.localNominated = true;
+
+    setTimeout(() => {
```

Comment:
argh, se me fue esto. en otro archivo usas 

```suggestion
    setTimeout(() => this.onHide(false), 1000);
```

y ese estilo me gusta más.